### ISSUE TITLE
refactor: move tests resources

### ISSUE THREAD

**Issue by surli** [INRIA/spoon#2137]
As pointed out in #2136 the tests directory is a bit messy, especially when you're not familiar with Spoon contribution. I would suggest to refactor it at least to have properly defined rules for the test resources.

We currently have test resources, in at least 3 different places:
  1. `src/test/resources` the standard directory
  2. `src/test/java/X/testclasses` in a subpackage, along with their test suite 
  3. `src/test/java/X` in the same package of the test suite

So I suggest first to move the test resources from case 3 to put - at least - the resources in a subpackage `testclasses`. 
I don't think it's desirable to move all test resources in place 1, as I think we want to check that we can compile them. 

WDYT?

**Comment by zielint0** [comment]
I am not yet completly familiary with Spoon terminology so please define test resource and give some examples (especially what can and should be changed, what must not be changed). Thank you.

**Comment by surli** [comment]
> Is it a good idea to mark every resource test class with Tc/TC prefix?

I'm not sure it's necessary if we put the classes in a proper package. 
And I'm almost sure that some class names are chosen to test some behaviours.

**Comment by zielint0** [comment]
Let's say I would like to move a test resource to proper location (testclasses):
- src/test/java/X -->
- --> src/test/java/X/testclasses

Should a test resource
spoon\src\test\java\spoon\support\compiler\jdt\ExtendedStringLiteralTestClass.java
be moved to
spoon\src\test\java\spoon\support\compiler\jdt\testclasses
?

Do I understand it correctly?

**Comment by surli** [comment]
> Do I understand it correctly?

Yes, exactly!

**Comment by zielint0** [comment]
@surli 
Simon please take a look on PR https://github.com/INRIA/spoon/pull/2147
Hope I recognized the correct ones.

**Comment by zielint0** [comment]
I moved some resources but got a [broken one check](http://spoon3r.lille.inria.fr/github-webhook//traces/tmpym3bdmo7) on Travis.
So closed https://github.com/INRIA/spoon/pull/2147
Will try to move them one by one in small PRs.

**Comment by zielint0** [comment]
@surli 
Should I move below files to testclasses?
- spoon/test/api/Metamodel.java
- spoon/test/api/AwesomeProcessor.java

**Comment by zielint0** [comment]
Status of refactoring - helps not to get lost
- [ ] Move ExtendedStringLiteralTestClass.java #2153
- [ ] Move ArrayClass.java to testclasses #2154

**Comment by zielint0** [comment]
@surli 
Most is moved. I thing there are few testclasses to move and we will close this soon.

**Comment by surli** [comment]
> Is it possible to get rid of: src/test/resources (for simplicity and compilation issues)?

uh? why would you want that? Those resources are used in our tests...

**Comment by zielint0** [comment]
Och, I don't mean deleting them physically :)
I mean to move them, so we can get simple layout:
- src/test/java/X
- src/test/java/X/testclasses

So only 2 places with tests, not 3.

**Comment by surli** [comment]
> So only 2 places with tests, not 3.
 
Actually in the `test/resources` there is no tests but only resources for the tests. We place there all resources that cannot be compiled but that can be used by Soon. Typically most resources there are related to `noclasspath` mode of Spoon.

So basically, they shouldn't be moved outside resources.

**Comment by zielint0** [comment]
Ok! Thank you so much for explanation!

**Comment by zielint0** [comment]
@surli 
I think most of testclasses are moved.
I will check tree again.
Meanwhile *refactor: move resources to testclasses in filters (re-introduced) #2222* is waiting.

**Comment by zielint0** [comment]
@surli 

I have moved some more - refactor: move resources to testclasses #2277
Question: do you qualify processors as test resources that should be moved to *testclasses* directory?
spoon/test/api/AwesomeProcessor.java

I don't think so, but I am not sure.

**Comment by surli** [comment]
No, processors are parts of the tests that are executed. They're not test resources.

**Comment by zielint0** [comment]
@surli 
@pvojtechovsky 
Is it safe to move VisitorFake (not a unused class!) to testclasses?
https://github.com/INRIA/spoon/pull/2278

**Comment by surli** [comment]
> Is it safe to move VisitorFake (not an unused class!) to testclasses?

I don't think it's a good idea: `VisitorFake` is a test, not an usual one but still, and not a test resource: so it should remain in the test package.

**Comment by zielint0** [comment]
@surli 
Thank you for clarification

**Comment by zielint0** [comment]
@surli 
> No, processors are parts of the tests that are executed. They're not test resources.

I moved misplaced processors outside test resources:
refactor: move processor classes to processors directory #2303
